state_estimation_node:
  ros__parameters:
    state_publish_rate: 0.001
    kalman_update_rate: 0.01
    state_lf_publish_rate: 0.5
    laying_on_ground_detection_offset: [ 0.0, 0.0, 0.0, 0.0 ]

    contact_detection:
      use_measured_forces: true
      force_threshold: [ 500.,500.,500.,500. ]
      energy_obs_kd: 20.
      energy_obs_threshold: 0.3
      leg_in_motion_joint_velocity_threshold: .08
      use_energy_obs_contact_detection: false
      energy_obs_filter_size: 20
      leg_swing_time: 0.2

    kalman:
      imu_measures_g: false
      assume_planar_ground: false
      adapt_covariances: false

      gyroscope_std: [ 0.005,0.004,0.0025 ]
      accelerometer_std: [ 0.15,0.2,0.4 ]
      gyroscope_bias_std: 0.001
      accelerometer_bias_std: 0.1
      contact_std: 0.0005
      belly_contact_point_measurement_std: 0.001
      foot_step_measurement_std: 0.03
      foot_on_plane_std: 0.01
      min_stance_percentage_for_contact: 0.1
      use_only_gait_contacts: true

      prior:
        gyroscope_bias: [ 0.,0.,0. ]
        accelerometer_bias: [ -0.004,0.005,0.04 ]

        base_orientation_std: 0.00001
        base_velocity_std: 0.00001
        base_position_std: 0.0000001
        gyroscope_bias_std: 0.01
        accelerometer_bias_std: 0.01





